TypeCobol/Compiler/Preprocessor/ReplaceTokensLinesIterator.cs

```diff
@@ -504,6 +509,78 @@ private Token CreateReplacedTokens(Token originalToken, ReplaceOperation replace
             }
         }
 
+        /// <summary>
+        /// Figure out a scan state from the given token.
+        /// </summary>
+        /// <param name="token">The Token to figure out the scan state.</param>
+        /// <returns>The Scan state in any, null otherwise</returns>
+        private static MultilineScanState FigureOutScanState(Token token)
+        {
+            if (token != null)
+            {
+                return token.ScanStateSnapshot ?? (token.TokensLine is TypeCobol.Compiler.Scanner.TokensLine tl
+                            && tl.ScanState != null ? tl.ScanState : null);
```

Comment:
Can be simplified as `return token.ScanStateSnapshot ?? (token.TokensLine is TypeCobol.Compiler.Scanner.TokensLine tl ? tl.ScanState : null);`

Reply:
ok done